### PR TITLE
Upgrade dynamodb label to `0.24.1`

### PR DESCRIPTION
**Pull request by nitrocode**
## what
* Upgrade dynamodb label to `0.24.1`
    * Bumped context for dynamodb from `0.22.0` to `0.24.1` which forced us to re-arrange `var.attributes` and `["lock"]` due to the change in `0.22.1`

## why
* Latest version of context label

## references
* https://github.com/cloudposse/terraform-null-label/releases/tag/0.22.1
* https://github.com/cloudposse/terraform-null-label/releases/tag/0.24.1
* 0.24.1 context pr https://github.com/cloudposse/terraform-aws-tfstate-backend/pull/82
* Closes https://github.com/cloudposse/terraform-aws-tfstate-backend/issues/91

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -167,8 +167,8 @@ resource "aws_s3_bucket_public_access_block" "default" {
 
 module "dynamodb_table_label" {
   source     = "cloudposse/label/null"
-  version    = "0.22.0"
-  attributes = compact(concat(var.attributes, ["lock"]))
+  version    = "0.24.1"
+  attributes = compact(concat(["lock"], var.attributes))
```
Comment (Nuru) [inline main.tf:171]:
```suggestion
  attributes =["lock"]
```

**Review by Nuru — CHANGES_REQUESTED** [review]

**Comment by nitrocode** [comment]
Since the random id generated is passed in via the `var.attributes` in the terratest, the `lock` portion now shows up after the random id instead of before. I'd like to update the test but I fear that this update will break backwards compatibility.

What do you think @Nuru ?

**Review by aknysh — APPROVED** [review]

**Pull request merged**